dask/local.py

```diff
@@ -463,11 +463,13 @@ def fire_tasks(chunksize):
                     ntasks = nready
                     chunksize = -(ntasks // -num_workers)
                 else:
-                    ntasks = min(nready, chunksize * num_workers)
+                    used_workers = -(len(state["running"]) // -chunksize)
+                    avail_workers = max(num_workers - used_workers, 0)
+                    ntasks = min(nready, chunksize * avail_workers)
```

Comment:
I don't understand chunksize here.  Do you understand the intent?  I would expect us to submit exactly as many tasks as we have workers.  I don't think that there is a need to speculatively submit things.

Reply:
(this is probably a question I should have asked back when this was added, my apologies for delaying that question until you're at the helm)

Reply:
Interesting.  OK, it looks like chunksize defaults to 1, so probably we're ok here.